server/fleet/calendar.go

```diff
@@ -41,6 +41,22 @@ type UserCalendar interface {
 	Get(event *CalendarEvent, key string) (interface{}, error)
 }
 
+// Lock interface for managing distributed locks.
+type Lock interface {
+	// AcquireLock attempts to acquire a lock with the given key. value is the value to set for the key, which is used to release the lock.
+	AcquireLock(ctx context.Context, key string, value string, expireMs uint64) (result string, err error)
+	// ReleaseLock attempts to release a lock with the given key and value. If key does not exist or value does not match, the lock is not released.
+	ReleaseLock(ctx context.Context, key string, value string) (ok bool, err error)
```

Comment:
@@Document that returns `true` if it was released, `false` otherwise (e.g. when `value` doesn't match).

Reply:
Will add